x-pack/legacy/plugins/ingest/server/libs/policy.contract.test.ts

```diff
@@ -0,0 +1,213 @@
+        );
+
+        servers = await createKibanaServer({
+          security: { enabled: true },
+        });
+        const esPolicy = JSON.parse(process.env.__JEST__ESServer || '');
+        es = new elasticsearch.Client({
+          hosts: esPolicy.hosts,
+          httpAuth: esPolicy.username ? `${esPolicy.username}:${esPolicy.password}` : undefined,
+        });
+      }
+    });
+
+    libs = compose(servers);
+    await libs.framework.waitForStack();
+  });
+
+  afterAll(async () => {
+    if (servers) {
+      await servers.shutdown();
+    }
+  });
+
+  beforeEach(async () => {
+    if (es) {
+      es.deleteByQuery({
+        index: INDEX_NAMES.INGEST,
+        body: {
+          conflicts: 'proceed',
+          query: { match_all: {} },
+        },
+      });
+    }
+  });
+
+  describe('create', () => {
+    it('should create a new policy', async () => {
+      const newPolicy = await libs.policy.create(TestUser, 'test', 'test description');
+
+      expect(typeof newPolicy.id).toBe('string');
+      expect(typeof newPolicy.shared_id).toBe('string');
+      expect(typeof newPolicy.version).toBe('number');
+
+      const gottenPolicy = await libs.policy.get(newPolicy.id);
+      expect(gottenPolicy.name).toBe('test');
+    });
+  });
+
+  describe('list', () => {
+    it('should list all active policies', async () => {
+      const newPolicy = await libs.policy.create(TestUser, 'test', 'test description');
+      const newPolicy2 = await libs.policy.create(TestUser, 'test2', 'test description');
+      const newPolicy3 = await libs.policy.create(TestUser, 'test3', 'test description');
+
+      expect(typeof newPolicy.id).toBe('string');
+      expect(typeof newPolicy.shared_id).toBe('string');
+      expect(typeof newPolicy.version).toBe('number');
+
+      const gottenPolicies = await libs.policy.list();
+      expect(gottenPolicies.length).toBe(3);
+      expect(gottenPolicies.find(c => c.id === newPolicy.id) !== undefined).toBe(true);
+      expect(gottenPolicies.find(c => c.id === newPolicy2.id) !== undefined).toBe(true);
+      expect(gottenPolicies.find(c => c.id === newPolicy3.id) !== undefined).toBe(true);
+    });
+
+    it('should not list inactive policies', async () => {
+      const newPolicy = await libs.policy.create(TestUser, 'test', 'test description');
+      const updated = await libs.policy.update(newPolicy.id, {
+        name: 'foo',
+      });
+      const newPolicy2 = await libs.policy.create(TestUser, 'test2', 'test description');
+      const newPolicy3 = await libs.policy.create(TestUser, 'test3', 'test description');
+
+      expect(typeof newPolicy.id).toBe('string');
+      expect(typeof newPolicy.shared_id).toBe('string');
+      expect(typeof newPolicy.version).toBe('number');
+
+      const gottenPolicies = await libs.policy.list();
+      expect(gottenPolicies.length).toBe(3);
+      expect(gottenPolicies.find(c => c.id === updated.id) !== undefined).toBe(true);
```

Comment:
`updated.id` id different than `newPolicy.id` ? if yes maybe add an expectation around that

Reply:
Yes, the updated policy has a new ID. Not sure what you mean about an exception though?

Reply:
@mattapperson not an exception but an expectation (an `expect` statement)